azext_edge/edge/commands_namespaces.py

```diff
@@ -355,6 +355,7 @@ def add_inbound_rest_device_endpoint(
     endpoint_name: str,
     endpoint_address: str,
     endpoint_version: Optional[str] = None,
+    certificate_reference: Optional[str] = None,
```

Comment:
For 2510, ADR actually has two new fields for X509 auth, which we need to add as parameters: `keySecretName` and `intermediateCertificatesSecretName`

Reference:
https://github.com/Azure/azure-rest-api-specs/blob/2caaf9e30d35dabfd6f24016b6ebe604f703287c/specification/deviceregistry/resource-manager/Microsoft.DeviceRegistry/stable/2025-10-01/deviceregistry.json#L8177

Reply:
This is not the same merged branch of the API spec: 
https://github.com/Azure/azure-rest-api-specs/pull/35911/files

Credential reference: https://github.com/Azure/azure-rest-api-specs/blob/2f8dcb350554bab45ef1e2ff35a05cfaf71b54e6/specification/deviceregistry/resource-manager/Microsoft.DeviceRegistry/stable/2025-10-01/deviceregistry.json#L8212

Iirc, these fields will not be in 2510 unless something changed recently.

Reply:
With @peolivei2's help, the main branch does have this change in there too https://github.com/Azure/azure-rest-api-specs/blob/main/specification/deviceregistry/resource-manager/Microsoft.DeviceRegistry/stable/2025-10-01/deviceregistry.json#L8173